src/solver/heuristic_board_solver.cpp

```diff
@@ -52,13 +52,15 @@ BoardState HeuristicBoardSolver::solve(const Board *board) const {
 void lookForOneColorLeftColumns(const Board *board, BoardState *state) {
   for (ushort column = 0; column < board->columnCount; ++column) {
     auto blanks = state->countColorInColumn(column, Blank);
-    ushort count0 = board->clueForColumn(column, C0).amount;
-    ushort count1 = board->clueForColumn(column, C1).amount;
 
-    if (count0 == blanks) {
-      paintBlanksOnColumn(board, state, column, C0);
-    } else if (count1 == blanks) {
-      paintBlanksOnColumn(board, state, column, C1);
+    for (auto color: board->colors) {
+      auto total = board->clueForColumn(column, color).amount;
+      auto count = state->countColorInColumn(column, color);
+
+      if (total - count == blanks) {
+        paintBlanksOnColumn(board, state, column, color);
+        break;
+      }
     }
   }
 }
```

Comment:
Line 69: why you need to cast to `ushort` the rowCount but the columnCount there is no need? 🤔